### PR TITLE
Report more BGZF errors

### PR DESCRIPTION
**Pull request by daviesrob**
* Make `bgzf_read_block()` print more error messages, including the offset of the BGZF block where it failed.

Note that this introduces a small change of behaviour as block_address is now updated after an empty block has been read.  This may change the location stored in indexes of files with empty blocks.

* Make `hts_itr_next()` return an error on seek failure.

The return code needs to be less that -1, or the caller will incorrectly interpret it as end of data, causing it to silently finish too early.

* Fix missing newline in tabix error messages.

### CODE REVIEW

**Comment by jmarshall** [comment]
There is a philosophy question here: should library functions print error messages themselves, or should they return -1 and anticipate that their callers will print a better-contextualised error message themselves?

Personally I prefer the latter, as the caller knows better what the high-level problem is, and otherwise you get a cascade of error messages from different levels pertaining to the same event. (OTOH messages from the library can give you the low-level technical details that aid debug tracing.)

TL;DR — perhaps your newly added `hts_log_error` calls would be better as `hts_log_debug` or so.

**Comment by daviesrob** [comment]
I'm going towards more logging, to avoid problems like #1037 where we now know something went wrong but have no idea what caused it.  This also (at last) prints the location in the file where the problem occurs, which is not something that the caller may have available to it - especially in the multi-threaded case.

**Comment by jkbonfield** [comment]
Ideally I'd favour not printing errors, but only *if* there is a proper hts_strerror and and hts_errno type model meaning the client tools can identify the actual source of error rather than a cryptic or misleading thing.

Long term I think overhauling error handling in that manner would be a win, but until it's done I favour printing them up in place - for now.

**Comment by jmarshall** [comment]
Hence the suggestion to make these `hts_log_debug()` — just like the existing one already in `bgzf_read_block()`. Best of both worlds IMHO.

**Comment by daviesrob** [comment]
These messages report actual errors when reading and decompressing the input files, so I think `hts_log_error()` is more appropriate than `hts_log_debug()`.  The messages will normally never be seen, but if there is a problem I want to know where it is without having to re-run a possibly lengthy process with all the debugging turned on.

If anything, we should remove some of the messages from the intermediate functions that don't add much information.  For example, the "Read block operation failed ..." message is rarely if ever useful as while it tells you something happened, it usually gives little clue as to what the problem was and where in the input it happened.
